lib/membrane_element_portaudio/sink.ex

```diff
@@ -63,20 +72,34 @@ defmodule Membrane.Element.PortAudio.Sink do
 
     endpoint_id = if endpoint_id == :default, do: @pa_no_device, else: endpoint_id
 
-    with {:ok, native} <-
+    with {:ok, {latency_ms, native}} <-
            SyncExecutor.apply(Native, :create, [
              self(),
+             ctx.clock,
              endpoint_id,
              ringbuffer_size,
              pa_buffer_size,
              latency
            ]) do
-      {:ok, %{state | native: native}}
+      {:ok, %{state | latency_time: latency_ms |> Time.milliseconds(), native: native}}
     else
       {:error, reason} -> {{:error, reason}, state}
     end
   end
 
+  @impl true
+  def handle_caps(:input, caps, _ctx, state) do
+    latency =
+      state.latency_time + Caps.frames_to_time((1.5 * state.portaudio_buffer_size) |> trunc, caps)
```

Comment:
I would prefer if it like this:
```elixir
latency =
(1.5 * state.portaudio_buffer_size)
|> trunc()
|> Caps.frames_to_time(...)
~> &1 + state.latency_time
```

~> can be replaced with non bunch equivalent lambda or |> Kernel.+(...) (which is a bit ugly)

Reply:
I'm looking at a bigger picture and it seems that `handle_caps` is not needed at all, so I'll remove it entirely